spec/controllers/attachments_controller_spec.rb

```diff
@@ -363,6 +363,57 @@ def expect_hidden(hidden_template)
     end
   end
 
+  context 'when the request is backpage' do
+    let(:prominence) { 'backpage' }
+    let(:incoming_message) { info_request.incoming_messages.first }
+
+    it 'sets a noindex header when viewing' do
+      get :show,
+          params: {
+            incoming_message_id: incoming_message.id,
+            id: info_request.id,
+            part: 2,
+            file_name: 'interesting.pdf',
+            skip_cache: 1
+          }
+      expect(response.headers['X-Robots-Tag']).to eq 'noindex'
+    end
+
+    it 'sets a noindex header when viewing a cached copy' do
```

Comment:
I added specs for viewing the cached copy because I'd initially included `ProminenceHeaders` _after_ the `cache_attachments` callback causing us to early return before `set_prominence_headers` got called.